tools/cli/domainCommands.go

```diff
@@ -807,6 +832,114 @@ func (d *domainCLIImpl) migrationDomainWorkFlowCheck(c *cli.Context) DomainMigra
 	}
 }
 
+func (d *domainCLIImpl) migrationDynamicConfigCheck(c *cli.Context) DomainMigrationRow {
+	var mismatchedConfigs []MismatchedDynamicConfig
+	check := true
+
+	resp := dynamicconfig.ListAllProductionKeys()
+
+	domain := c.GlobalString(FlagDomain)
+	newDomain := c.String(FlagDestinationDomain)
+	ctx, cancel := newContext(c)
+	defer cancel()
+
+	for _, configName := range resp {
+		if len(configName.Filters()) == 1 && configName.Filters()[0] == dc.DomainName {
+			// Use the helper method to get the GetDynamicConfigRequest
+			currRequest := dynamicconfig.ToGetDynamicConfigFilterRequest(configName.String(), []dynamicconfig.FilterOption{
+				dynamicconfig.DomainFilter(domain),
+			})
+
+			newRequest := dynamicconfig.ToGetDynamicConfigFilterRequest(configName.String(), []dynamicconfig.FilterOption{
+				dynamicconfig.DomainFilter(newDomain),
+			})
+
+			currResp, err := d.frontendAdminClient.GetDynamicConfig(ctx, currRequest)
+			if err != nil {
+				fmt.Println("Failed to fetch dynamic config for current domain:", err)
+			}
+			newResp, err := d.destinationAdminClient.GetDynamicConfig(ctx, newRequest)
+			if err != nil {
+				fmt.Println("Failed to fetch dynamic config for current domain:", err)
+			}
+
+			if currResp.Value != newResp.Value {
+				check = false
+				mismatchedConfig := MismatchedDynamicConfig{
+					CurrResp: currResp,
+					NewResp:  newResp,
+				}
+				mismatchedConfigs = append(mismatchedConfigs, mismatchedConfig)
+			}
+
+		} else if len(configName.Filters()) == 1 && configName.Filters()[0] == dc.DomainID {
+
+			currentDomainID := d.getDomainID(domain, ctx)
+			if currentDomainID == "" {
+				fmt.Println("Failed to get domainID for the current domain.")
+
+			}
+
+			destinationDomainID := d.getDomainID(newDomain, ctx)
+			if destinationDomainID == "" {
+				fmt.Println("Failed to get domainID for the destination domain.")
+			}
+
+			currRequest := dynamicconfig.ToGetDynamicConfigFilterRequest(configName.String(), []dynamicconfig.FilterOption{
+				dynamicconfig.DomainIDFilter(currentDomainID),
+			})
+
+			newRequest := dynamicconfig.ToGetDynamicConfigFilterRequest(configName.String(), []dynamicconfig.FilterOption{
+				dynamicconfig.DomainIDFilter(destinationDomainID),
+			})
+
+			currResp, err := d.frontendAdminClient.GetDynamicConfig(ctx, currRequest)
+			if err != nil {
+				fmt.Println("Failed to fetch dynamic config for the current domain.")
```

Comment:
instead of printing the error, use empty to annotate it's not available `currResp = &types.GetDynamicConfigResponse{}`